test/y2network/config_test.rb

```diff
@@ -146,4 +159,18 @@
       end
     end
   end
+
+  describe "#rename_interface" do
+    it "adjusts the interface name" do
+      config.rename_interface("eth0", "eth1", :mac)
+      eth1 = config.interfaces.by_name("eth1")
+      expect(eth1.renaming_mechanism).to eq(:mac)
+    end
+
+    it "adjusts the connection configurations for that interface" do
+      config.rename_interface("eth0", "eth1", :mac)
+      eth1_conns = config.connections.by_interface("eth1")
+      expect(eth1_conns).to_not be_empty
+    end
```

Comment:
can you just for sure add test for tricky double rename? that was eth0, then eth1 and then eth2? In past we also have issues with renames of multiple interfaces and with editing renamed interface, but it is something we should test manually.